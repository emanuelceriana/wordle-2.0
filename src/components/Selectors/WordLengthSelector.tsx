import cn from "classnames";
import { useState } from "react";
import wordsDS from "../../data/wordsDataSet.json";
import { useCreateObjectList } from "../hooks/useCreateObjectList";
import styles from "./WordLengthSelector.module.scss";

interface WordLengthSelectorProps {
  defaultValue: number;
  onClickRequest: (key: number) => void;
}

export const WordLengthSelector = ({
  defaultValue,
  onClickRequest,
}: WordLengthSelectorProps) => {
  const [optionSelected, setOptionSelected] = useState<number>(defaultValue);

  const { objectList: wordLengthButtons } = useCreateObjectList({
    length: Object.keys(wordsDS).length,
  });

  const handleClickAction = (key: string) => {
    setOptionSelected(parseInt(key));
    onClickRequest(parseInt(key));
  };

  return (
    <div>
      <div className={styles.wordLengthSelector}>
        <span>Word Length:</span>
        <div className={styles["wordLengthSelector__buttons"]}>
          {wordLengthButtons.map(({ id }, idx) => {
            const key = Object.keys(wordsDS)[idx];
            return (
              <button
                key={id}
                className={cn(styles["wordLengthSelector__button"], {
                  [styles["wordLengthSelector__selected"]]:
                    optionSelected.toString() === key,
                })}
                onClick={() => handleClickAction(key)}
              >
                {key}
              </button>
            );
          })}
        </div>
      </div>
    </div>
  );
};

export default WordLengthSelector;
