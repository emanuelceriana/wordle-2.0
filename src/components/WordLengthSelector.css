.wordLengthSelector {
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 2em;
}

.wordLengthSelector span {
  line-height: 2em;
  margin-right: 0.5em;
}

.wordLengthSelector .buttons {
  display: flex;
  justify-content: center;
  align-items: center;
}

.wordLengthSelector .selected {
  border: 2px solid white;
}

.wordLengthSelector .button {
  width: 3em;
  height: 3em;
  font-size: 0.7em;
  margin: 0.2em;
  display: flex;
  justify-content: center;
  align-items: center;
}
