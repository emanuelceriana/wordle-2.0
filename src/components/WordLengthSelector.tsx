import cn from "classnames";
import { useState } from "react";
import wordsDS from "../data/wordsDataSet.json";
import "./WordLengthSelector.css";
import { useCreateObjectList } from "./hooks/useCreateObjectList";

interface WordLengthSelectorProps {
  defaultValue: string;
  onClickRequest: (key: number) => void;
}

export const WordLengthSelector = ({
  defaultValue,
  onClickRequest,
}: WordLengthSelectorProps) => {
  const [optionSelected, setOptionSelected] = useState<string>(defaultValue);

  const { objectList: wordLengthButtons } = useCreateObjectList({
    length: Object.keys(wordsDS).length,
  });

  const handleClickAction = (key: string) => {
    setOptionSelected(key);
    onClickRequest(parseInt(key));
  };

  return (
    <div>
      <div className="wordLengthSelector">
        <span>Word Length:</span>
        <div className="buttons">
          {wordLengthButtons.map(({ id }, idx) => {
            const key = Object.keys(wordsDS)[idx];
            return (
              <button
                key={id}
                className={cn("button", { selected: optionSelected === key })}
                onClick={() => handleClickAction(key)}
              >
                {key}
              </button>
            );
          })}
        </div>
      </div>
    </div>
  );
};

export default WordLengthSelector;
