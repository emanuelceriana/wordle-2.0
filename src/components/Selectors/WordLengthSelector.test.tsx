import { render, fireEvent } from "@testing-library/react";
import { WordLengthSelector } from "./WordLengthSelector";

describe("<WordLengthSelector />", () => {
  const defaultValue = 5;
  const setDefaultValue = jest.fn();
  it("should render without errors", () => {
    const { asFragment } = render(
      <WordLengthSelector
        defaultValue={defaultValue}
        onClickRequest={setDefaultValue}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("should render with default value selected", () => {
    const { container } = render(
      <WordLengthSelector
        defaultValue={defaultValue}
        onClickRequest={setDefaultValue}
      />
    );

    expect(
      container.querySelector(".wordLengthSelector__selected")?.innerHTML
    ).toBe(defaultValue.toString());
  });
  it("should change value correctly", () => {
    const newValue = 7;
    const { getByText } = render(
      <WordLengthSelector
        defaultValue={defaultValue}
        onClickRequest={setDefaultValue}
      />
    );

    fireEvent.click(getByText(newValue.toString()));

    expect(setDefaultValue).toHaveBeenCalledWith(newValue);
  });
});
