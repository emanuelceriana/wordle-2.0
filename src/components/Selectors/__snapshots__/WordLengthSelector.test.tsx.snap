// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WordLengthSelector /> should render without errors 1`] = `
<DocumentFragment>
  <div>
    <div
      class="wordLengthSelector"
    >
      <span>
        Word Length:
      </span>
      <div
        class="wordLengthSelector__buttons"
      >
        <button
          class="wordLengthSelector__button"
        >
          4
        </button>
        <button
          class="wordLengthSelector__button wordLengthSelector__selected"
        >
          5
        </button>
        <button
          class="wordLengthSelector__button"
        >
          6
        </button>
        <button
          class="wordLengthSelector__button"
        >
          7
        </button>
        <button
          class="wordLengthSelector__button"
        >
          8
        </button>
        <button
          class="wordLengthSelector__button"
        >
          9
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
